### PR TITLE
Add security headers to the default response 

### LINKED ISSUE #111: Add security headers to prevent XSS
Some browsers such as Internet Explorer require the ​nosniff header to be set and potentially dangerous characters to be encoded. Otherwise other websites embedding this resource could trigger a XSS vulnerability.

### PR DESCRIPTION
**Pull request by xh3n1**
Added security headers such as:
  ```
'X-Content-Type-Options' =>'nosniff',
  'Content-Security-Policy' => "default-src 'none'",
   'X-Frame-Options' => 'DENY'
```
Fix #111

### CODE REVIEW

**Comment by xh3n1** [comment]
@oliverklee Thank you for the review 😄 
I made the changes, and actually wrote a system test for the SecuredViewHandler. Also fixed #112

File: .travis.yml [hunk 1]
```diff
@@ -73,6 +73,11 @@ script:
   echo "Running the integration tests";
   vendor/bin/phpunit tests/Integration/;
 
+- >
+  echo;
+  echo "Running the system tests";
+  vendor/bin/phpunit tests/System/;
```
Comment (oliverklee) [inline .travis.yml]: @xh3n1 Thanks for the change.

I'd like to stick to the "one change/topic per PR" policy. Could you please move this to a separate PR? This will keep the history (and git blame) clean and allow for clean reverts if needed. Thanks!

Comment (xh3n1) [reply .travis.yml]: @oliverklee Thanks, I added it as a separate PR.

File: config/services.yml [hunk 2]
```diff
@@ -4,3 +4,11 @@ services:
...
         autowire: true
         tags: ['controller.service_arguments']
+
+    my.secure_handler:
+      class: \PhpList\RestBundle\ViewHandler\SecuredViewHandler
+
+    my.secure_view_handler:
+      parent: fos_rest.view_handler.default
+      calls:
+       - ['registerHandler', [ 'json', ["@my.secure_handler", 'createResponse'] ] ]
```
Comment (oliverklee) [inline config/services.yml]: Please let's unify this to always use single quotes unless the double quotes are required here.

File: src/ViewHandler/SecuredViewHandler.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     * @param Request     $request
+     * @param string      $format
+     *
+     * @return Response
+     */
+    public function createResponse(ViewHandler $handler, View $view, Request $request, string $format)
+    {
+        $view->setHeaders([
+            'X-Content-Type-Options' => 'nosniff',
+            'Content-Security-Policy' => "default-src 'none'",
```
Comment (oliverklee) [inline src/ViewHandler/SecuredViewHandler.php]: Would it also work to switch the single and double quotes here?

File: src/ViewHandler/SecuredViewHandler.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This class is used to add headers to the default response.
+ * @author Xheni Myrtaj <xheni@phplist.com> .
+ */
+class SecuredViewHandler
+{
+
```
Comment (oliverklee) [inline src/ViewHandler/SecuredViewHandler.php]: Please let's drop the blank line here.

File: src/ViewHandler/SecuredViewHandler.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+namespace PhpList\RestBundle\ViewHandler;
+
+use FOS\RestBundle\View\View;
+use FOS\RestBundle\View\ViewHandler;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This class is used to add headers to the default response.
+ * @author Xheni Myrtaj <xheni@phplist.com> .
```
Comment (oliverklee) [inline src/ViewHandler/SecuredViewHandler.php]: Let's polish this: :-)

Could you please add a blank line above the `@author` line and drop the period at the end of the `@author` line?

File: src/ViewHandler/SecuredViewHandler.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+     * @param ViewHandler $viewHandler
+     * @param View        $view
+     * @param Request     $request
+     * @param string      $format
+     *
+     * @return Response
+     */
+    public function createResponse(ViewHandler $handler, View $view, Request $request, string $format)
+    {
+        $view->setHeaders([
```
Comment (oliverklee) [inline src/ViewHandler/SecuredViewHandler.php]: Please let's put the `[` and `]` on separate lines. This will avoid formatting changes when using the PhpStorm autoformat or the php-cs-fixer.

File: tests/System/Controller/SecuredViewHandlerTest.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
+<?php
+declare(strict_types=1);
+namespace PhpList\RestBundle\Tests\System\Controller;
```
Comment (oliverklee) [inline tests/System/Controller/SecuredViewHandlerTest.php]: To keep things consistent, please let's have a blank line between the `declare` line and the `namespace` section.

File: tests/System/Controller/SecuredViewHandlerTest.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+use GuzzleHttp\Client;
+use PhpList\Core\TestingSupport\Traits\SymfonyServerTrait;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Test for security headers
+ * @author Xheni Myrtaj <xheni@phplist.com>
+ *
+ *
```
Comment (oliverklee) [inline tests/System/Controller/SecuredViewHandlerTest.php]: Please let's remove the blank lines.

File: tests/System/Controller/SecuredViewHandlerTest.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+namespace PhpList\RestBundle\Tests\System\Controller;
+
+use GuzzleHttp\Client;
+use PhpList\Core\TestingSupport\Traits\SymfonyServerTrait;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Test for security headers
+ * @author Xheni Myrtaj <xheni@phplist.com>
```
Comment (oliverklee) [inline tests/System/Controller/SecuredViewHandlerTest.php]: For consistency, please let's have a blank line above the `@author` line.

File: tests/System/Controller/SecuredViewHandlerTest.php [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ *
+ */
+class SecuredViewHandlerTest extends TestCase
+{
+
```
Comment (oliverklee) [inline tests/System/Controller/SecuredViewHandlerTest.php]: Please let's drop the blank line. (In general, there should be no blank line below a `{` or above a `}`.)

File: tests/System/Controller/SecuredViewHandlerTest.php [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+        $this->startSymfonyServer($environment);
+
+        $response = $this->httpClient->get(
+            '/api/v2/sessions',
+            ['base_uri' => $this->getBaseUrl()]
+        );
+        $expectedHeaders = [
+            'X-Content-Type-Options' => 'nosniff',
+            'Content-Security-Policy' => "default-src 'none'",
+            'X-Frame-Options' => 'DENY'
```
Comment (oliverklee) [inline tests/System/Controller/SecuredViewHandlerTest.php]: Please let's have a comma after the last array element.

**Comment by xh3n1** [comment]
Thanks for the review @oliverklee I applied the requested coding style and rebased to master. I hope that I didn't miss anything. Just a question about changelog.md, under which release should I add the changes?

**Comment by oliverklee** [comment]
> Just a question about changelog.md, under which release should I add the changes?

Under "x.y.z (next release)", please.

File: src/ViewHandler/SecuredViewHandler.php [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /**
+     * @param ViewHandler $viewHandler
+     * @param View $view
+     * @param Request $request
+     * @param string $format
+     *
+     * @return Response
+     */
+    public function createResponse(ViewHandler $handler, View $view, Request $request, string $format)
```
Comment (oliverklee) [inline src/ViewHandler/SecuredViewHandler.php]: Would it work to have `Response` as a return type declaration?

**Comment by xh3n1** [comment]
ok thanks@oliverklee, I will create another PR for that.

File: src/ViewHandler/SecuredViewHandler.php (hunk 12 above)
Comment (xh3n1) [reply src/ViewHandler/SecuredViewHandler.php]: Yes, it works.

**Comment by oliverklee** [comment]
> I will create another PR for that.

Optimally, the changelog entry for a PR should be part of the same PR.

File: src/ViewHandler/SecuredViewHandler.php [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+{
+    /**
+     * @param ViewHandler $viewHandler
+     * @param View $view
+     * @param Request $request
+     * @param string $format
+     *
+     * @return Response
+     */
+    public function createResponse(ViewHandler $handler, View $view, Request $request, string $format):Response
```
Comment (oliverklee) [inline src/ViewHandler/SecuredViewHandler.php]: Almost there! :-)

Could you please add a space after the `:`?

And now that we have all types as type declarations, the PHPDoc does not offer any additional information, and we can remove it (assuming that PHPCS won't complain).

Comment (oliverklee) [reply src/ViewHandler/SecuredViewHandler.php]: Then let's keep it for this PR. For the mid-term, I'd like to remove all PHPDoc comments that only repeat the PHP (return) type declaration and add nothing, hence reducing the amount of code that needs to be maintained.
